mardi_importer/mardi_importer/publications/ZenodoResource.py

```diff
@@ -105,6 +110,51 @@ def resource_type(self):
                 self._resource_type = "wd:Q37866906"
         return self._resource_type
 
+    @property
+    def communities(self):
+        if not self._communities:
+            for communityCur in self.metadata["communities"]:
+                community_id = communityCur.get("id")
+                if not community_id == "mathplus":
+                    next
+                community = Community(api = self.api, community_id = community_id)
+                community_qid = community.create()
+                self._communities.append(community)
+        return self._communities
+    
+    @property
+    def projects(self):
+        community = None
+        if self._communities:
+            for communityCur in self._communities:
+                if communityCur.community_id == "mathplus":
+                    community = communityCur
+                    break
+        if (not self._projects) and community and self.metadata.get("related_identifiers"):
+            #print(Project.get_project_ids())
+            for related_ids in self.metadata.get("related_identifiers"):
+                print("identifier: " + related_ids["identifier"])
+                if related_ids["identifier"] in Project.get_project_ids():
+                    project = Project(api = self.api, community = community, project_id = related_ids["identifier"])
+                    #if project.exis
+                    projet_QID = project.create()
+                    self._projects.append(project)
+        return self._projects
+
+    def exists(self):
+        
+        if self.QID:
+            return self.QID
+
+        QID_results = self.api.search_entity_by_value("wdt:P4901", str(self.zenodo_id))
```

Comment:
I think this block between lines 149 and 157 can be deleted because it runs already once in __post_init__, assigning the value of self.QID if one is found. If a QID is not found in __post_init__, then there won't be a value for QID, which means that the item does not exist. In that second case, there is no need to run the search again. If line 146 is not successful, exists() will return False by default.